python/GafferUI/GLWidget.py

```diff
@@ -249,6 +246,88 @@ def keyPressEvent( self, event ) :
 
 		QtWidgets.QFrame.keyPressEvent( self, event )
 
+	@classmethod
+	def __createGLWidget( cls, format ) :
+
+		# try to make a host specific widget if necessary.
+		result = cls.__createMayaQGLWidget( format )
+		if result is not None :
+			return result
+
+		result = cls.__createHoudiniQGLWidget( format )
+		if result is not None :
+			return result
+
+		glWidget = QtWidgets.QOpenGLWidget()
+		# Avoid `QOpenGLFramebufferObject: Framebuffer incomplete attachment`
+		# errors caused by Qt trying to make a framebuffer with zero size.
+		glWidget.setMinimumSize( 1, 1 )
+		glWidget.setFormat( format )
+
+		return glWidget
+
+	@classmethod
+	def __createQGLFormat( cls, format ):
+		"""
+		Create a QGLFormat based on the configuration of
+		QSurfaceFormat where possible.
+		"""
+		qGLFormat = QtOpenGL.QGLFormat()
+		qGLFormat.setRgba( True )
```

Comment:
I realised why you need this - because it has nothing to do with choosing alpha vs no alpha. Instead it's about avoiding a crazy old indexed colour mode. I think this implies that you do still need an additional `if format.hasAlpha()` conditional where you call `qGLFormat.setAlpha( True )`.

Reply:
I'm using `hasAlpha` to sync the two formats and I brought back the `setDepth` as well.